xbmc/video/dialogs/GUIDialogAudioSubtitleSettings.cpp

```diff
@@ -293,9 +293,13 @@ void CGUIDialogAudioSubtitleSettings::InitializeSettings()
     g_application.m_pPlayer->GetSubtitleCapabilities(m_subCaps);
   }
 
+  // register IsPlayingPassthrough condition
+  m_settingsManager->AddCondition("IsPlayingPassthrough", IsPlayingPassthrough);
+
   CSettingDependency dependencyAudioOutputPassthroughDisabled(SettingDependencyTypeEnable, m_settingsManager);
-  dependencyAudioOutputPassthroughDisabled.And()
-    ->Add(CSettingDependencyConditionPtr(new CSettingDependencyCondition(SETTING_AUDIO_PASSTHROUGH, "false", SettingDependencyOperatorEquals, false, m_settingsManager)));
+  dependencyAudioOutputPassthroughDisabled.Or()
```

Comment:
Because without it the volume and volume amplification setting weren't automatically updated when the passthrough setting was changed. But I now realized that this doesn't cover changing the audio stream. I tried to manually update the enabled state after the audio stream setting was changed but by then the player hadn't loaded the new stream yet and `IsPassthrough()` still returned the value of the old stream. Do you have any idea how this worked before?